core/aws_ddk_core/stages/athena_sql.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional
+
+from aws_cdk.aws_events import EventPattern, IRuleTarget, RuleTargetInput
+from aws_cdk.aws_events_targets import SfnStateMachine
+from aws_cdk.aws_kms import Key
+from aws_cdk.aws_s3 import Location
+from aws_cdk.aws_stepfunctions import IntegrationPattern, StateMachine, StateMachineType, Succeed
+from aws_cdk.aws_stepfunctions_tasks import (
+    AthenaStartQueryExecution,
+    EncryptionConfiguration,
+    EncryptionOption,
+    QueryExecutionContext,
+    ResultConfiguration,
+)
+from aws_ddk_core.pipelines import DataStage
+from aws_ddk_core.resources import StepFunctionsFactory
+from constructs import Construct
+
+
+class AthenaSQLStage(DataStage):
+    """
+    Class that represents a Athena SQL DDK DataStage.
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        environment_id: str,
+        query_string: str,
+        workgroup: Optional[str] = None,
+        catalog_name: Optional[str] = None,
+        database_name: Optional[str] = None,
+        output_bucket_name: Optional[str] = None,
+        ouput_object_key: Optional[str] = None,
+        encryption_option: Optional[EncryptionOption] = None,
+        encryption_key: Optional[Key] = None,
+        state_machine_input: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        DDK Athena SQL stage.
+
+        Stage that contains a step function that executre Athena SQL query.
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the stage
+        environment_id : str
+            Identifier of the environment
+        query_string : str
+            SQL query that will be started
+        state_machine_input : Optional[Dict[str, Any]]
+            Input of the state machine
```

Comment:
Looks to missing docs for some of the params..

Reply:
Thanks